File: Java/AnagramDetector.java
import java.util.Arrays;

public class AnagramDetection {
    public static boolean isAnagram(String str1, String str2) {
        char[] s1 = str1.toCharArray();
        char[] s2 = str2.toCharArray();

        Arrays.sort(s1);
        Arrays.sort(s2);

        return Arrays.equals(s1, s2);
    }

    public static void main(String[] args) {
        String str1 = "listen";
        String str2 = "silent";

        if (isAnagram(str1, str2)) {
            System.out.println(str1 + " and " + str2 + " are anagrams.");
        } else {
            System.out.println(str1 + " and " + str2 + " are not anagrams.");
        }
    }
}

File: Rust/AnagramDetector.rs
fn is_anagram(str1: &str, str2: &str) -> bool {
    let mut s1 = str1.chars().collect::<Vec<_>>();
    let mut s2 = str2.chars().collect::<Vec<_>>();

    s1.sort();
    s2.sort();

    s1 == s2
}

fn main() {
    let str1 = "listen";
    let str2 = "silent";

    if is_anagram(str1, str2) {
        println!("{} and {} are anagrams.", str1, str2);
    } else {
        println!("{} and {} are not anagrams.", str1, str2);
    }
}